frontend/src/components/NewChecklist/Option/OptionButton/OptionButton.tsx

```diff
@@ -25,19 +23,31 @@ const OptionButton = ({ option }: { option: OptionWithIcon }) => {
       text: theme.palette.grey500,
       fill: theme.palette.white,
     },
-    selected: { border: theme.palette.yellow600, fill: theme.palette.yellow100, text: theme.palette.yellow700 },
+    selected: {
+      border: theme.palette.yellow600,
+      fill: theme.palette.yellow100,
+      text: theme.palette.yellow700,
+    },
   };
 
-  const currentColor = selectedOptionActions.isSelectedOption(id) ? BUTTON_COLOR.selected : BUTTON_COLOR.unSelected;
+  const currentColor = isSelected ? BUTTON_COLOR.selected : BUTTON_COLOR.unSelected;
 
   return (
     <S.Box color={currentColor.fill} borderColor={currentColor.border} onClick={handleClick}>
-      <S.IconBox>{selectedOptionActions.isSelectedOption(id) ? <FilledIcon /> : <UnFilledIcon />}</S.IconBox>
+      <S.IconBox>{isSelected ? <FilledIcon /> : <UnFilledIcon />}</S.IconBox>
       <S.TextBox color={currentColor.text}>{displayName}</S.TextBox>
     </S.Box>
   );
 };
 
+/*사용자가 누른 옵션 버튼만 리렌더링*/
+const MemoizedOptionButton = React.memo((props: { option: OptionWithIcon; isSelected: boolean }) => {
+  const isSelected = useSelectedOptionStore(state => state.actions.isSelectedOption(props.option.id));
+  return <OptionButton option={props.option} isSelected={isSelected} />;
+});
+
+export default MemoizedOptionButton;
```

Comment:
prop으로 isSelected를 받고있는데 정작 isSelected를 사용하지는 않고, 
내부에서 새로 지역변수를 만들어서 사용하고있네요.

지금의 사용례라면 그냥 이 6줄 전체를 아래와 같이 대체할 수 있을 것 같아요

```suggestion
export default React.memo(OptionButton);
```